### ISSUE TITLE
Code: QCBM classes for the addon package

### ISSUE THREAD

**Issue by nrhawkins** [QCHPC/qiskit_QCBMs#5]
**Create classes that can be used to create a QCBM, when someone pip installs the qiskit_addon_qcbm package.**

1. The class needs to allow user choices for: ansatz, loss function, distance metric function, cl optimizer, and any config params for creating and running the code to train the QCBM.

2.  Maybe the Qiskit Code Assistant can be used to help create the code: https://github.com/Qiskit/qiskit-code-assistant-vscode/blob/main/LOCAL_SETUP.md
3. Design used by other projects could be examined for comparison, ideas about what the interface could look like.
